internal/service/fsx/openzfs_file_system.go

```diff
@@ -0,0 +1,844 @@
+			},
+			"subnet_ids": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MinItems: 1,
+				MaxItems: 1,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"throughput_capacity": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntInSlice([]int{64, 128, 256, 512, 1024, 2048, 3072, 4096}),
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"weekly_maintenance_start_time": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(7, 7),
+					validation.StringMatch(regexp.MustCompile(`^[1-7]:([01]\d|2[0-3]):?([0-5]\d)$`), "must be in the format d:HH:MM"),
+				),
+			},
+		},
+
+		CustomizeDiff: verify.SetTagsDiff,
+	}
+}
+
+func resourceOepnzfsFileSystemCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).FSxConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	input := &fsx.CreateFileSystemInput{
+		ClientRequestToken: aws.String(resource.UniqueId()),
+		FileSystemType:     aws.String(fsx.FileSystemTypeOpenzfs),
+		StorageCapacity:    aws.Int64(int64(d.Get("storage_capacity").(int))),
+		StorageType:        aws.String(d.Get("storage_type").(string)),
+		SubnetIds:          flex.ExpandStringList(d.Get("subnet_ids").([]interface{})),
+		OpenZFSConfiguration: &fsx.CreateFileSystemOpenZFSConfiguration{
+			DeploymentType: aws.String(d.Get("deployment_type").(string)),
+		},
+	}
+
+	backupInput := &fsx.CreateFileSystemFromBackupInput{
+		ClientRequestToken: aws.String(resource.UniqueId()),
+		StorageType:        aws.String(d.Get("storage_type").(string)),
+		SubnetIds:          flex.ExpandStringList(d.Get("subnet_ids").([]interface{})),
+		OpenZFSConfiguration: &fsx.CreateFileSystemOpenZFSConfiguration{
+			DeploymentType: aws.String(d.Get("deployment_type").(string)),
+		},
+	}
+
+	if v, ok := d.GetOk("disk_iops_configuration"); ok {
+		input.OpenZFSConfiguration.DiskIopsConfiguration = expandFsxOpenzfsFileDiskIopsConfiguration(v.([]interface{}))
+		backupInput.OpenZFSConfiguration.DiskIopsConfiguration = expandFsxOpenzfsFileDiskIopsConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("root_volume_configuration"); ok {
+		input.OpenZFSConfiguration.RootVolumeConfiguration = expandFsxOpenzfsRootVolumeConfiguration(v.([]interface{}))
+		backupInput.OpenZFSConfiguration.RootVolumeConfiguration = expandFsxOpenzfsRootVolumeConfiguration(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		input.KmsKeyId = aws.String(v.(string))
+		backupInput.KmsKeyId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("automatic_backup_retention_days"); ok {
```

Comment:
i would put this inline in the `OpenZFSConfiguration` because GetOk will skip zero value and zero value is default

Reply:
done